File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> res(2u);
        int low = 0;
        int high = numbers.size() - 1;
        
        while(low < high){
            int current_sum = numbers[low] + numbers[high];
            
            if (current_sum == target){
                res[0u] = low + 1;
                res[1u] = high + 1;
            }
            
            if (current_sum > target){
                high--;
            }
            else{
                low++;
            }
        }
        
        
        return res;
    }
};

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int right = numbers.length - 1;
        int left = 0;

        while (left < right){
            int temp_sum = numbers[left] + numbers[right]; 
            
            if (target == temp_sum) {
                break;
            }
            
            if (target < temp_sum){
                right--;
            }
            else{
                left++;
            }
        }

        return IntStream.of(left + 1, right + 1).toArray();
    }
}